### PR TITLE
Add clear history button

### LINKED ISSUE #1015: [Feature Request] add "clear console" button to console menu
Was testing some things on extremely-alpha and I noticed a feature that would be nice to have: an ability to clear the console. I don't know if there are system level commands in iodide, but a `clear` command would be great.

### PR DESCRIPTION
**Pull request by mdboom**
Fixes #1015 and #1090 

image

### CODE REVIEW

**Comment by hamilton** [comment]
I think the position of that tooltip could be place so that it appears in the lower right side of the icon. At the moment, the editor pane is overlapping a tooltip that is actually appearing in the eval frame. I'm not sure how we address this with this current setup, but it does seem like one solution is to just prevent the tooltip from being in the overlapping area to the best of our abilities.

Setting the tooltip placement ([outlined here](https://material-ui.com/demos/tooltips/#positioned-tooltips)) to `bottom-end` might solve the issue.

**Comment by mdboom** [comment]
Oh, CSS is such a joy!  I'll poke at it and let you know if I get stuck.

**Comment by mdboom** [comment]
I think I've fixed the tooltip and shrinking height issues.

**Comment by hamilton** [comment]
LGTM - thanks @mdboom!

File: src/eval-frame/components/panes/console-toolbar.jsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+          >
+            <DeleteButton fontSize="small" />
+          </IconButton>
+        </Tooltip>
+      </MiniToolbar>
+    )
+  }
+}
+
+export function mapStateToProps() {
```
Comment (bcolloran) [inline src/eval-frame/components/panes/console-toolbar.jsx]: this mapStateToProps is not needed
https://stackoverflow.com/questions/47657365/can-i-mapdispatchtoprops-without-mapstatetoprops-in-redux

File: src/eval-frame/components/panes/console-toolbar.jsx [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  return { }
+}
+
+function mapDispatchToProps(dispatch) {
```
Comment (bcolloran) [inline src/eval-frame/components/panes/console-toolbar.jsx]: preferred method is to just select out the actions you need, as in:
https://github.com/iodide-project/iodide/blob/3d14fb7a32af15e50f2dcddf6b0042a65710c478/src/components/cells/cell-container.jsx#L154

we have this pattern still floating around, but it is legacy

File: src/components/controls/mini-toolbar.jsx [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+      </Toolbar>
+    )
+  }
+}
+
+export function mapStateToProps() {
+  return { }
+}
+
+const MiniToolbar = connect(mapStateToProps)(MiniToolbarUnconnected)
```
Comment (bcolloran) [inline src/components/controls/mini-toolbar.jsx]: this does not appear to need to be connected to the redux store at all, right? it's just a simple presentational component?

**Comment by hamilton** [comment]
Thanks for the more thorough review @bcolloran. I think I dropped the ball on this PR review + testing more thoroughly on Chrome.

**Comment by mdboom** [comment]
Sorry about this, all.  I still feel like a monkey on a typewriter with React.  Is there a good reference / book?

**Comment by bcolloran** [comment]
No worries @mdboom!! :-) We're here to help you through it. Unfortunately we have a bunch a legacy code from when we were first learning that demonstrates bad patterns, and in many places it's not a great foundation to learn from... so really it's me and hamilton who must apologize to you!

I don't know of any particular books, but i think the official react and redux docs are excellent. I've also found quite a number of great hackernoon and medium articles that have helped a lot when googling around specific topics.

**Pull request merged**